dask/dataframe/utils.py

```diff
@@ -434,7 +434,11 @@ def check_matching_columns(meta, actual):
         if extra or missing:
             extra_info = f"  Extra:   {extra}\n  Missing: {missing}"
         else:
-            extra_info = "Order of columns does not match"
+            extra_info = (
+                f"Order of columns does not match."
+                f"\nActual:   { { c: str(t) for c, t in meta.dtypes.items() } }"
+                f"\nExpected: { { c: str(t) for c, t in actual.dtypes.items() } }"
```

Comment:
Why include dtype info here and not just the order of the columns? Seems irrelevant for the error being raised

Reply:
Good question. I like to see what types of function generals are for sanity check purposes. 

Also, I probably don't understand the purpose of this exception. If we know the order of the columns, why not simply reorder the columns?

Reply:
@jrbourbeau, let's convert `dict` to `list` if you think it makes more sense.

Reply:
Yeah, let's do that. I think just listing the columns in order makes the most sense here.